File: reto 2/app/src/main/java/com/example/androidtictactoe/MainActivity.java
package com.example.androidtictactoe;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class MainActivity extends AppCompatActivity {

    EditText mEditText;
    Button mButton;

    String playerName = "";

    FirebaseDatabase database;
    DatabaseReference databaseReference;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mEditText = findViewById(R.id.editText);
        mButton = findViewById(R.id.button);

        database = FirebaseDatabase.getInstance();

        // Check if player exist and get reference
        SharedPreferences preferences = getSharedPreferences("PREFS", 0);
        playerName = preferences.getString("playerName", "");
        if (!playerName.equals("")) {
            databaseReference = database.getReference("players/" + playerName);
            addEventListener();
            databaseReference.setValue("");
        }

        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Loggin the player in
                playerName = mEditText.getText().toString();
                mEditText.setText("");
                if (!playerName.equals("")) {
                    mButton.setText("Logging in");
                    mButton.setEnabled(false);
                    databaseReference = database.getReference("players/" + playerName);
                    addEventListener();
                    databaseReference.setValue("");
                }
            }
        });
    }

    private void addEventListener() {
        // Read from database
        databaseReference.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {
                // Success - continue next screen after saving the player name
                if (!playerName.equals("")) {
                    SharedPreferences preferences = getSharedPreferences("PREFS", 0);
                    SharedPreferences.Editor editor = preferences.edit();
                    editor.putString("playerName", playerName);
                    editor.apply();

                    startActivity(new Intent(getApplicationContext(), MainActivity2.class));
                    finish();
                }
            }

            @Override
            public void onCancelled(@NonNull DatabaseError error) {
                // Error
                mButton.setText("Log in");
                mButton.setEnabled(true);
                Toast.makeText(MainActivity.this, "Error", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: reto7/app/src/main/java/com/example/challenge8/MainActivity.kt
package com.example.challenge8

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun passToCreateCompany(view: View){
        val intent = Intent(this, MainActivity2::class.java)
        startActivity(intent)
    }

    fun passToListCompany(view: View){
        val intent = Intent(this, CompanyList::class.java)
        startActivity(intent)
    }

    fun passToFilterByNameCompany(view: View){
        val intent = Intent(this, NameFilter::class.java)
        startActivity(intent)
    }

    fun passToFilterByCategory(view: View){
        val dataBundle = Bundle()
        dataBundle.putBoolean("searchForCategory", true)
        val intent = Intent(this, NameFilter::class.java)
        intent.putExtras(dataBundle)
        startActivity(intent)
    }

}